raiden/network/resolver/client.py

```diff
@@ -41,16 +42,31 @@ def reveal_secret_with_resolver(
         "expiration": secret_request_event.expiration,
         "payment_recipient": to_hex(raiden.address),
         "chain_id": chain_state.chain_id,
-        "chain_height": chain_state.block_number,
     }
 
-    try:
-        response = requests.post(raiden.config["resolver_endpoint"], json=request)
-    except requests.exceptions.RequestException:
-        return False
+    # loop until we get a valid response from the resolver or until timeout
+    while True:
+        current_state = raiden.wal.state_manager.current_state
+        assert isinstance(current_state, ChainState)
 
-    if response is None or response.status_code != HTTPStatus.OK:
-        return False
+        if secret_request_event.expiration < current_state.block_number:
+            return False
+
+        response = None
+
+        try:
+            # before calling resolver, update block height
+            request["chain_height"] = chain_state.block_number
+            response = requests.post(raiden.config["resolver_endpoint"], json=request)
+        except requests.exceptions.RequestException:
+            pass
+
+        if response is not None:
+            if response.status_code == HTTPStatus.OK:
+                break
+            if response.status_code == HTTPStatus.NOT_FOUND:
```

Comment:
XUD will also sometimes return HTTP 400 in case of a bad resolve request or HTTP 500 in case of an internal error that caused the swap to fail, so I think you should probably be checking for these too (especially the 500 code which I believe should happen with some regularity). Or you can do `else` in any case where HTTPStatus is not OK.

Reply:
What should raiden do in case of 400 or 500? These are bugs that should be handled by a developer and there is no "right" handling.

Anyway, this was already merged into raiden's develop branch. If you think we should let raiden continue it's flow in case of 400 and 500 errors please convert these to 404 xud side.

Reply:
Oh I was just saying that raiden should treat it as a failure and `return False` like it does on the next line. I do see it was already merged into raiden's develop branch, let me see if I can propose those changes there.